extractor/src/main/java/org/schabi/newpipe/extractor/services/youtube/extractors/YoutubeStreamExtractor.java

```diff
@@ -809,28 +909,24 @@ private void checkPlayabilityStatus(final JsonObject youtubePlayerResponse,
                             "This age-restricted video cannot be watched.");
                 }
             }
-            if (status.equalsIgnoreCase("unplayable")) {
-                if (reason != null) {
-                    if (reason.contains("Music Premium")) {
-                        throw new YoutubeMusicPremiumContentException();
-                    }
-                    if (reason.contains("payment")) {
-                        throw new PaidContentException("This video is a paid video");
-                    }
-                    if (reason.contains("members-only")) {
-                        throw new PaidContentException(
-                                "This video is only available for members of the channel of this video");
-                    }
-                    if (reason.contains("unavailable")) {
-                        final String detailedErrorMessage = getTextFromObject(playabilityStatus
-                                .getObject("errorScreen").getObject("playerErrorMessageRenderer")
-                                .getObject("subreason"));
-                        if (detailedErrorMessage != null) {
-                            if (detailedErrorMessage.contains("country")) {
-                                throw new GeographicRestrictionException(
-                                        "This video is not available in user's country.");
-                            }
-                        }
+            if (status.equalsIgnoreCase("unplayable") && reason != null) {
+                if (reason.contains("Music Premium")) {
+                    throw new YoutubeMusicPremiumContentException();
+                }
+                if (reason.contains("payment")) {
+                    throw new PaidContentException("This video is a paid video");
+                }
+                if (reason.contains("members-only")) {
+                    throw new PaidContentException(
+                            "This video is only available for members of the channel of this video");
+                }
+                if (reason.contains("unavailable")) {
+                    final String detailedErrorMessage = getTextFromObject(playabilityStatus
+                            .getObject("errorScreen").getObject("playerErrorMessageRenderer")
+                            .getObject("subreason"));
+                    if (detailedErrorMessage != null && detailedErrorMessage.contains("country")) {
+                        throw new GeographicRestrictionException(
+                                "This video is not available in user's country.");
```

Comment:
Isn't the `else` missing here? So `"unplayable"` and `"unavailable"` videos without a `detailedErrorMessage` involving country availability would pass, instead of throwing an exception as they ought to.

Reply:
No, they should not pass because of the `ContentNotAvailableException` 5 lines below, thrown for every case when `status` is not null and different of `OK`.

Reply:
Ok, sorry